msgpack/src/Data/MessagePack/Object.hs

```diff
@@ -90,6 +95,14 @@ data Object
     -- __NOTE__: MessagePack is limited to maximum extension data size of up to \( 2^{32}-1 \) bytes.
   deriving (Show, Read, Eq, Ord, Typeable, Generic)
 
+(.:) :: MessagePack a => [(Object, Object)] -> T.Text -> Result a
```

Comment:
could also be an `ObjectMap` instead of a `[(Object, Object)]` which would mean using a `Vector` lookup instead.

For larger objects it may be quicker to create a `Map` and lookup in there instead.